python/lsst/ctrl/bps/bps_draw.py

```diff
@@ -38,97 +38,9 @@ def draw_networkx_dot(graph, outname):
-            tcnt += 1
-            tnode_name = ".".join(nodes.taskDef.taskName.split(".")[-2:])
-
-            ofh.write("task%d [shape=none, margin=0, label=<\n" % (tcnt))
-            ofh.write('<table border="0" cellborder="1" cellspacing="0" cellpadding="4">\n')
-            ofh.write(
-                '<TR><TD><b>TD</b></TD><TD colspan="%d">%s</TD></TR>\n' % (len(nodes.quanta), tnode_name)
-            )
-
-            # write quantum headers
-            ofh.write("<TR><TD><b>Q</b></TD>")
-            colcnt = 0
-            for q_id in range(1, len(nodes.quanta) + 1):
-                qcnt += 1
-                colcnt += 1
-                if colcnt % 2 == 0:
-                    ofh.write("<TD><b>Q%02d</b></TD>" % qcnt)
-                else:
-                    ofh.write('<TD BGcolor="lightgrey"><b>Q%02d</b></TD>' % qcnt)
-            ofh.write("</TR>\n")
-
-            # write quantum inputs
-            ofh.write("<TR><TD><b>IN</b></TD>")
-            colcnt = 0
-            for quantum in nodes.quanta:
-                dr_ids = []
-                for ds_refs in quantum.predictedInputs.values():
-                    for ds_ref in ds_refs:
-                        ds_name = "%s+%s" % (ds_ref.datasetType.name, ds_ref.dataId)
-                        if ds_name not in dsname_to_node_id:
-                            drcnt += 1
-                            dsname_to_node_id[ds_name] = drcnt
-                        dr_ids.append("dr%03d" % dsname_to_node_id[ds_name])
-                colcnt += 1
-                if colcnt % 2 == 0:
-                    ofh.write("<TD>%s</TD>" % ",".join(dr_ids))
-                else:
-                    ofh.write('<TD BGcolor="lightgrey">%s</TD>' % ", ".join(dr_ids))
-            ofh.write("</TR>\n")
-
-            # write quantum outputs
-            ofh.write("<TR><TD><b>OUT</b></TD>")
-            colcnt = 0
-            for quantum in nodes.quanta:
-                dr_ids = []
-                for ds_refs in quantum.outputs.values():
-                    for ds_ref in ds_refs:
-                        ds_name = "%s+%s" % (ds_ref.datasetType.name, ds_ref.dataId)
-                        if ds_name not in dsname_to_node_id:
-                            drcnt += 1
-                            dsname_to_node_id[ds_name] = drcnt
-                        dr_ids.append("dr%03d" % dsname_to_node_id[ds_name])
-                colcnt += 1
-                if colcnt % 2 == 0:
-                    ofh.write("<TD>%s</TD>" % ",".join(dr_ids))
-                else:
-                    ofh.write('<TD BGcolor="lightgrey">%s</TD>' % ", ".join(dr_ids))
-            ofh.write("</TR>\n")
-            ofh.write("</table>>];\n")
-
-        # add invisible edges so force vertical
-        for i in range(1, tcnt):
-            ofh.write("task%d -> task%d;" % (i, i + 1))
-        ofh.write("}\n")
-
-    _LOG.info("tasks=%d dataset refs=%d", tcnt, drcnt)
+    networkx.drawing.nx_pydot.write_dot(graph, outname)
```

Comment:
This draw function is actually for the ScienceDAG and WorkflowGraph (it had more code in it when I could easily get pygraphviz in my local stack).  The Pipeline graph and QuantumGraph drawing is now made via the pipetask command line options.  I do want to have all the drawing style bits, etc in this function instead of in the graphs themselves.  I am treating this as really low priority at the moment because it is probably only used by me when debugging and for demos.  If we make any interesting changes to it, I'll make sure to ping you.